zigbeetradfri/integrationpluginzigbeetradfri.cpp

```diff
@@ -637,6 +714,67 @@ void IntegrationPluginZigbeeTradfri::initOnOffSwitch(ZigbeeNode *node, ZigbeeNod
     });
 }
 
+void IntegrationPluginZigbeeTradfri::initShortcutButton(ZigbeeNode *node, ZigbeeNodeEndpoint *endpoint)
+{
+    // Get the current configured bindings for this node
+    ZigbeeReply *reply = node->removeAllBindings();
+    connect(reply, &ZigbeeReply::finished, node, [=](){
+        if (reply->error() != ZigbeeReply::ErrorNoError) {
+            qCWarning(dcZigbeeTradfri()) << "Failed to remove all bindings for initialization of" << node;
+        } else {
+            qCDebug(dcZigbeeTradfri()) << "Removed all bindings successfully from" << node;
+        }
+
+        // Read battery, bind and configure attribute reporting for battery
+        if (!endpoint->hasInputCluster(ZigbeeClusterLibrary::ClusterIdPowerConfiguration)) {
+            qCWarning(dcZigbeeTradfri()) << "Failed to initialize the power configuration cluster because the cluster could not be found" << node << endpoint;
+            return;
+        }
+
+        qCDebug(dcZigbeeTradfri()) << "Read power configuration cluster attributes" << node;
+        ZigbeeClusterReply *readAttributeReply = endpoint->getInputCluster(ZigbeeClusterLibrary::ClusterIdPowerConfiguration)->readAttributes({ZigbeeClusterPowerConfiguration::AttributeBatteryPercentageRemaining});
```

Comment:
This whole block could be replaced by calling `bindPowerConfigrationCluster(endpoint)`

Reply:
although the plugin would need to be changed to inherit from ZigbeeIntegration plugin and with that some little changes in setupThing() are needed, as well es dropping thingRemoved() and nodeRemoved() so the base class handles that on its own.